ui/app/src/components/PathNavigatorTree/PathNavigatorTree.tsx

```diff
@@ -153,14 +155,22 @@ export default function PathNavigatorTree(props: PathNavigatorTreeProps) {
   const [rootNode, setRootNode] = useState(null);
 
   const hasActiveSession = useSelection((state) => state.auth.active);
+  const intervalRef = useRef<any>();
+
+  const resetBackgroundRefreshInterval = useCallback(() => {
+    clearInterval(intervalRef.current);
+    intervalRef.current = setInterval(() => {
+      dispatch(pathNavigatorTreeBackgroundRefresh({ id }));
+    }, backgroundRefreshTimeoutMs);
+  }, [backgroundRefreshTimeoutMs, dispatch, id]);
 
   useEffect(() => {
     if (backgroundRefreshTimeoutMs && hasActiveSession) {
-      let interval = setInterval(() => {
+      intervalRef.current = setInterval(() => {
```

Comment:
Since we made resetBackgroundRefreshInterval a callback, we can use it on this effect too and avoid the code duplicity